doc/adr/0006-protocol-naming-conventions.md

```diff
@@ -0,0 +1,55 @@
+# 6. protocol-naming-conventions
+
+Date: 2021-05-08
+
+## Status
+
+Under Review
+
+## Context
+
+This document describes naming conventions for these protocol components:
+
+* Subjects (including Reply Subjects)
+* Stream Names
+* Consumer Names
+* Account Names
+
+## Prior Work
+
+Currently the NATS Docs regarding [protocol convention](https://docs.nats.io/nats-protocol/nats-protocol#protocol-conventions) says this:
+
+> Subject names, including reply subject (INBOX) names, are case-sensitive and must be non-empty alphanumeric strings with no embedded whitespace. All ascii alphanumeric characters except spaces/tabs and separators which are "." and ">" are allowed. Subject names can be optionally token-delimited using the dot character (.), e.g.:
+A subject is comprised of 1 or more tokens. Tokens are separated by "." and can be any non space ascii alphanumeric character. The full wildcard token ">" is only valid as the last token and matches all tokens past that point. A token wildcard, "*" matches any token in the position it was listed. Wildcard tokens should only be used in a wildcard capacity and not part of a literal token.
+
+> Character Encoding: Subject names should be ascii characters for maximum interoperability. Due to language constraints and performance, some clients may support UTF-8 subject names, as may the server. No guarantees of non-ASCII support are provided.
+
+## Specification
+
+```
+dot              = "."
+asterisk         = "*"
+lt               = "<"
+gt               = ">"
+dollar           = "$"
+colon            = ":"
+double-quote     = ["]
+fwd-slash        = "/"
+backslash        = "\"
+pipe             = "|"
+question-mark    = "?"
+ampersand        = "&"
+printable        = all printable ascii (33 to 126 inclusive)
+term             = (printable except dot, asterisk or gt)+
+prefix           = (printable except dot, asterisk, gt or dollar)+
+filename-safe    = (printable except dot, asterisk, lt, gt, colon, double-quote, fwd-slash, backslash, pipe, question-mark, ampersand)
+
+message-subject    = term (dot term | asterisk)* (dot gt)?
+reply-to           = term (dot term)*
+stream-name        = term
+queue-name         = term
+durable-name       = term
+js-internal-prefix = dollar (prefix dot)+
+js-user-prefix     = (prefix dot)+
+account-name       = (filename-safe)+ maximum 255 characters
```

Comment:
Realistically speaking no this can not  be 255 characters, windows allow max 255 for the entire path name+path should not exceed 255 so a 255 character account name would not be usable on windows machines as servers

Reply:
Correct, windows has issues. Windows server previous to 2016 maxes at 260 characters, after that it's 1024. Windows 10 defaults to 260, but can be configured to support 32k. I think it's reasonable to limit the account-name to something much shorter, maybe 16, 24 or 32 characters. Any opinions on this length?

Reply:
At least length of Nkey which is 57 or so? ~60 IIRC.